### PR TITLE
Make WMTS LegendURL configurable

### PR DESCRIPTION
**Pull request by simonseyock**
Adds two new config parameters for layers: `wmts_kvp_legendurl` and `wmts_rest_legendurl`. If those are set, MapProxy will not try to generate a LegendURL on its own (if `legendurl` was provided), but will instead use this one (no matter if the `legendurl` was provided or not).

### CODE REVIEW

File: doc/configuration.rst [hunk 1]
```diff
@@ -392,6 +392,20 @@ Each dimension is another dictionary with a list of ``values`` and an optional `
             - 1000
             - 3000
 
+``wmts_kvp_legendurl`` and ``wmts_rest_legendurl``
+""""""""""""""""""""""""""""""""""""""""""""""""""
+
+It is possible which URLS the WMTS will advertise as LegendURLs through its capabilities. These will be inserted as a
```
Comment (weskamm) [inline doc/configuration.rst:398]:
```suggestion
It is possible to configure which URLS the WMTS will advertise as LegendURLs through its capabilities. These will be inserted as a
```

File: doc/configuration.rst [hunk 2]
```diff
@@ -392,6 +392,20 @@ Each dimension is another dictionary with a list of ``values`` and an optional `
...
+``LegendURL`` tag into the capabilities document. These URLs support two template variables ``{base_url}`` which is the
+base url of MapProxy and ``{layer_name}`` which is the name of the layer.
+
+.. code-block:: yaml
+
+  layers:
+    - name: legend_layer
+      tile_sources: [cache]
+      wmts_rest_legendurl: "{base_url}/legend/{layer_name}.png" # would become "http://127.0.0.1/legend/osm.png" on localhost
```
Comment (weskamm) [inline doc/configuration.rst:407]:
```suggestion
      wmts_rest_legendurl: "{base_url}/legend/{layer_name}.png" # would become "http://127.0.0.1/legend/legend_layer.png" on localhost
```

**Review by weskamm — COMMENTED** [review]
Can you add an configuration / outcome example for wmts_kvp_legendurl ?

File: doc/configuration.rst (hunk 2 above)
Comment (simonseyock) [reply doc/configuration.rst:407]: nice catch

**Review by weskamm — APPROVED** [review]

**Pull request merged**